power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -170,21 +167,277 @@ template <bool sym> class NewtonRaphsonSESolver {
     SparseLUSolver<NRSEGainBlock<sym>, NRSERhs<sym>, NRSEUnknown<sym>> sparse_solver_;
     typename SparseLUSolver<NRSEGainBlock<sym>, NRSERhs<sym>, NRSEUnknown<sym>>::BlockPermArray perm_;
 
-    void prepare_matrix_and_rhs(YBus<sym> const& /*y_bus*/, MeasuredValues<sym> const& /*measured_value*/,
-                                ComplexValueVector<sym> const& /*current_u*/) {
+    void prepare_matrix_and_rhs(YBus<sym> const& y_bus, MeasuredValues<sym> const& measured_value,
```

Comment:
the cognitive complexity of this function is out of the park. please split it up

Reply:
comments need to be made on that PR as well before closing this one